### PR TITLE
Get type information from function annotations

### PR DESCRIPTION
**Pull request by acetylen**
This PR relates to #46. It replaces `inspect.getfullargspec` with `inspect.signature` (or `funcsigs.signature` if below python 3.3) to allow easy access to annotations, and then uses them as a fallback if type information is not available in the function docstring.

I've added four test cases to check that this works and that providing arguments that can't convert properly results in an exception.

I've also inadvertently re-ordered some imports. My linters *do not like* the code style in this project :smile:

### CODE REVIEW

**Comment by acetylen** [comment]
Oh, it's still trying to test against the deprecated 3.4, and PyYAML has dropped support. I was worried I'd broken something.

**Comment by acetylen** [comment]
Is there a reason for keeping 2.6 compatibility around? Since Travis can't even build it, I mean.

**Comment by rubik** [comment]
@acetylen 2.6 can be removed if we want. I'd like to keep 2.7 around for a while still. There shouldn't be any problems with that version on Travis.

**Comment by acetylen** [comment]
There we go. I noticed in local testing that pypy3 works just fine too, so I added that to the list of tests.

File: mando/core.py [hunk 1]
```diff
@@ -125,27 +118,38 @@ def _generate_command(self, func, name=None, doctype='rest',
         subparser.set_defaults(**{_DISPATCH_TO: func})
         return func
 
-    def _analyze_func(self, func, params, argz, varargs_name):
+    def _analyze_func(self, func, doc_params):
         '''Analyze the given function, merging default arguments, overridden
         arguments (with @arg) and parameters extracted from the docstring.
 
         :param func: The function to analyze.
         :param params: Parameters extracted from docstring.
```
Comment (rubik) [inline mando/core.py]: It should be `doc_params`.

**Comment by rubik** [comment]
@acetylen Great! There is a minor change needed in the code. Then we need a couple of sentences in the README and in the documentation. Since you did all the work, I'd like to credit you in the changelog. Currently it does not exist for Mando, but we can copy it from Radon:
https://github.com/rubik/radon/blob/master/CHANGELOG

You can add your name following the format in that one. The next version will be 0.7.0.

File: mando/core.py (hunk 1 above)
Comment (acetylen) [reply mando/core.py]: Could have swore I fixed that...

**Comment by acetylen** [comment]
I think that should be everything. I left the changelog blank below 0.7.0 because I don't know the details of previous releases.

File: README.rst [hunk 2]
```diff
@@ -1,35 +1,41 @@
...
-    :alt: Mando license
-    :target: https://pypi.python.org/pypi/mando/
 
 mando is a wrapper around ``argparse``, and allows you to write complete CLI
 applications in seconds while maintaining all the flexibility.
 
 Installation
 ------------
 
-Mando is tested across all Python versions from **Python 2.6** to **Python
```
Comment (acetylen) [inline README.rst]: In case it's unclear, I added badges that relay this information (pyversions and implementation).

**Comment by rubik** [comment]
@acetylen Great, perfect! Just make one last change: in the CHANGELOG you can remove the previous version and TODO. We will start the CHANGELOG from this version. I would do it myself but this branch is in your repository.

**Comment by rubik** [comment]
@acetylen Thanks again for all the work, I merged the changes and I'll release a new version soon.

**Comment by rubik** [comment]
@acetylen I published version 0.7.0.

**Comment by acetylen** [comment]
Thank you for accepting, now I have a go-to parser for any future projects!

**Comment by rubik** [comment]
@acetylen Great! If you come across new features that could be useful let's discuss.

**Pull request merged**